src/index.js

```diff
@@ -1,34 +1,95 @@
 const mysql = require('mysql');
+const http = require('http');
+const { Server } = require('socket.io');
 const app = require('./app');
 const config = require('./config/config');
 const logger = require('./config/logger');
+const { EVENTS } = require('./config/events');
 
-let server;
+const { createMessage } = require('./services/chat.service');
+const { getUserNameById } = require('./services/user.service');
 
-async function start() {
+const server = http.createServer(app);
+const io = new Server(server, { cors: { origin: '*', credentials: true } });
+
+// let server;
 
+async function start() {
   try {
-    const connection = mysql.createConnection({
+    const dbConnection = mysql.createConnection({
       host: process.env.MYSQL_HOST,
       user: process.env.MYSQL_USERNAME,
       password: process.env.MYSQL_PASSWORD,
-      database: process.env.MYSQL_DATABASE
-    })
+      database: process.env.MYSQL_DATABASE,
+    });
 
-    connection.connect();
+    dbConnection.connect();
     logger.info('Connected to MySql DB');
+    logger.info('Sockets enabled');
+
+    const connectedUsers = {};
+
+    io.on(EVENTS.connection, (socket) => {
+      logger.info(`user connected ${socket.id}`);
+
+      /* Register connected user */
+      socket.on(EVENTS.CLIENT.REGISTER, async (id) => {
+        const userId = Number(id);
+
+        /** You can check phpAuth() here and put the below code in ifelse */
+
+        const name = await getUserNameById(userId);
 
-    server = app.listen(config.port, () => {
+        socket.userId = userId; // eslint-disable-line no-param-reassign
+        socket.userName = name.user_name; // eslint-disable-line no-param-reassign
+        connectedUsers[userId] = socket;
+
+        const ActiveUsers = Object.values(connectedUsers).map((user) => ({
+          userId: user.userId,
+          userName: user.userName,
+        }));
+
+        io.sockets.emit(EVENTS.SERVER.USERS, ActiveUsers);
+      });
```

Comment:
Every time a user registers and gets added, this event will notify all the users who are listening to this event. 
It doesn't make sense right ?